plugins/guests/linux/cap/mount_smb_shared_folder.rb

```diff
@@ -76,6 +79,21 @@ def self.mount_smb_shared_folder(machine, name, guestpath, options)
 
           emit_upstart_notification(machine, expanded_guest_path)
         end
+
+        def self.merge_mount_options(base, overrides)
+          base = base.join(",").split(",")
+          overrides = overrides.join(",").split(",")
+          b_kv = Hash[base.map{|item| item.split("=", 2) }]
+          o_kv = Hash[overrides.map{|item| item.split("=", 2) }]
+          merged = {}.tap do |opts|
+            (b_kv.keys + o_kv.keys).uniq.each do |key|
+              opts[key] = o_kv.fetch(key, b_kv[key])
+            end
+          end
+          merged.map do |key, value|
+            [key, value].compact.join("=")
+          end
+        end
```

Comment:
We should have some tests around this function

Reply:
This provides implicit testing https://github.com/hashicorp/vagrant/pull/9366/files/b1c38c282e13550dead1bfa9805e6a60a3487a4a#diff-8ea7a458713d53670287098e3ad15a5bR96

but some explicit tests can be provided too.